src/agent/onefuzz-agent/src/tasks/coverage/generic.rs

```diff
@@ -188,6 +189,28 @@ impl<'a> TaskContext<'a> {
     }
 
     pub async fn record_input(&mut self, input: &Path) -> Result<()> {
+        let attempts = MAX_COVERAGE_RECORDING_ATTEMPTS;
+
+        for _ in 0..attempts {
+            if let Err(err) = self.try_record_input(input).await {
```

Comment:
Can we keep the result here, and just add context stating it failed after N attempts as this functions Result::Err?